### PR TITLE
Add BadgeView Control

### PR DESCRIPTION
**Pull request by tnassour**
### Description of Change ###

 

 ### Linked Issues ###
 

 - Fixes #

 ### PR Checklist ###
 
 - [ ] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [ ] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls/XYZ 

 ### Additional information ###

### CODE REVIEW

**Comment by brminnick** [comment]
Assigning [Proposal](https://github.com/CommunityToolkit/Maui/issues/105) Champion, @pictos

File: samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs
Comment (VladislavAntonyuk) [inline samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs:27]: Use ICommand attribute instead

File: src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs:328]: is null

File: src/CommunityToolkit.Maui/Views/BadgeView/IBadgeAnimation.shared.cs
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Views/BadgeView/IBadgeAnimation.shared.cs:6]: It should be in Core

**Review by VladislavAntonyuk — CHANGES_REQUESTED** [review]

**Comment by brminnick** [comment]
Thanks @tnassour! 

I just wanted to give you a heads up that we'll need you to open a PR for our [Docs repo](https://github.com/MicrosoftDocs/CommunityToolkit) before we can merge this PR: https://github.com/MicrosoftDocs/CommunityToolkit

For now, I've added the `pending documentation` and `do not merge` labels so that we don't forget 👍

File: samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj
Comment (tnassour) [reply samples/CommunityToolkit.Maui.Sample/CommunityToolkit.Maui.Sample.csproj]: Removed

File: samples/CommunityToolkit.Maui.Sample/Pages/Views/BadgeViewPage.xaml
Comment (tnassour) [reply samples/CommunityToolkit.Maui.Sample/Pages/Views/BadgeViewPage.xaml]: Changed

File: samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs
Comment (tnassour) [reply samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs]: modified

**Comment by bijington** [comment]
@tnassour is this PR something that you are still able to look at? We are on a feature freeze until .NET MAUI GAs later this month but I would like to see this go into the next version. If you are unable it's no problem, I would be happy to assist in getting it completed.

File: src/CommunityToolkit.Maui/Views/BaseTemplatedView.shared.cs
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Views/BaseTemplatedView.shared.cs:34]:
```suggestion
		if (Control is null && child is TControl control)
```

**Review by VladislavAntonyuk — CHANGES_REQUESTED** [review]

File: src/CommunityToolkit.Maui/Views/BadgeView/IBadgeAnimation.shared.cs
Comment (tnassour) [reply src/CommunityToolkit.Maui/Views/BadgeView/IBadgeAnimation.shared.cs]: moved

File: src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs
Comment (tnassour) [reply src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs]: done

File: src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs
Comment (tnassour) [reply src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs]: changed

File: src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs
Comment (tnassour) [reply src/CommunityToolkit.Maui/Views/BadgeView/BadgeView.shared.cs:220]: nope cause the label uses both FontFamily, FontAttributes and the Font property is only a getter

File: samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs
Comment (tnassour) [reply samples/CommunityToolkit.Maui.Sample/ViewModels/Views/BadgeViewViewModel.cs]: done

**Comment by tnassour** [comment]
> @tnassour is this PR something that you are still able to look at? We are on a feature freeze until .NET MAUI GAs later this month but I would like to see this go into the next version. If you are unable it's no problem, I would be happy to assist in getting it completed.

@bijington i have done the modifications, it remains testing the sample solution for all of this.
While building locally i am getting this error : 
`3>CSC : error CS0016: Could not write to output file 'C:\Apps\CommunityToolKit\Maui-fork\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_9e00a6b2-8703-4c7f-aaa2-32f7268ecea1.g.shared.cs' -- 'Could not find a part of the path 'C:\Apps\CommunityToolKit\Maui-fork\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_9e00a6b2-8703-4c7f-aaa2-32f7268ecea1.g.shared.cs'.'`

can someone tell me how to fix this ?

**Comment by VladislavAntonyuk** [comment]
> > @tnassour is this PR something that you are still able to look at? We are on a feature freeze until .NET MAUI GAs later this month but I would like to see this go into the next version. If you are unable it's no problem, I would be happy to assist in getting it completed.
> 
> @bijington i have done the modifications, it remains testing the sample solution for all of this. While building locally i am getting this error : `3>CSC : error CS0016: Could not write to output file 'C:\Apps\CommunityToolKit\Maui-fork\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_9e00a6b2-8703-4c7f-aaa2-32f7268ecea1.g.shared.cs' -- 'Could not find a part of the path 'C:\Apps\CommunityToolKit\Maui-fork\samples\CommunityToolkit.Maui.Sample\obj\GeneratedFiles\CommunityToolkit.Maui.SourceGenerators\CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator\SearchBarTextColorTo_9e00a6b2-8703-4c7f-aaa2-32f7268ecea1.g.shared.cs'.'`
> 
> can someone tell me how to fix this ?

https://github.com/CommunityToolkit/Maui/issues/340#issuecomment-1081052363

File: src/CommunityToolkit.Maui/Views/BaseTemplatedView.shared.cs
Comment (VladislavAntonyuk) [inline src/CommunityToolkit.Maui/Views/BaseTemplatedView.shared.cs:7]: TemplatedView uses Compatibility StackLayout. I think we shouldn't use controls from Compatibility namespace

**Comment by brminnick** [comment]
Closing PR as stale + abandoned

**Pull request closed without merging**
